### PR TITLE
Extract `Dsl` generator from `Generator` class

### PR DESCRIPTION
**Pull request by mutecipher**
### Motivation

We want to extract each generator from the current Generator, because:

better design: later we'll be able to introduce a common ancestor to all generators
easier testing: we will be able to unit test each generator in isolation
better separation of concerns: by breaking the coupling between generators

### Implementation

* Extract `build_dsl` from `Generator` class into isolated class
* Decouple parameters from `Config` file
* Update `Cli` to use new `Dsl` class

### Tests

See automated tests (no changes).

### CODE REVIEW

File: lib/tapioca/config.rb [hunk 1]
```diff
@@ -29,6 +29,7 @@ def outpath
     TAPIOCA_CONFIG = T.let("#{TAPIOCA_PATH}/config.yml", String)
 
     DEFAULT_COMMAND = T.let("bin/tapioca", String)
+    DEFAULT_COMPILER_PATH = T.let("#{__dir__}/compilers/dsl", String)
```
Comment (mutecipher) [inline lib/tapioca/config.rb:32]: As mentioned in a previous PR, this is a static path that never changes, and was being dynamically generated in the `Generator` class. Moved it here as it makes more sense and won't be affected by the file moving to a different namespace.

File: lib/tapioca/generators.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# typed: true
```
Comment (mutecipher) [inline lib/tapioca/generators.rb:1]: Moved to a top-level `generators` namespace to make for easier updates in the future. Can be updated in the future to dynamically pull in all generators.

**Review by paracycle — APPROVED** [review]
Looks good to me

File: lib/tapioca/config.rb (hunk 1 above)
Comment (Morriar) [reply lib/tapioca/config.rb:32]: I'm not sure about this one:
* Having it under `Config` implies this is something configurable.
* Having `default` in the name implies this is a default value that may be changed.
* It doesn't say anything about `dsl`

What do you think about moving it under `Tapioca::Compilers::Dsl::COMPILERS_PATH`?

**Pull request merged**